crates/forge/src/multi_runner.rs

```diff
@@ -391,8 +395,25 @@ impl MultiContractRunnerBuilder {
                 deployable_contracts.insert(id.clone(), (abi.clone(), bytecode, libs_to_deploy));
             }
 
-            if let Some(bytes) = linked_contract.get_deployed_bytecode_bytes() {
-                known_contracts.insert(id.clone(), (abi.clone(), bytes.to_vec()));
+            if !use_zk {
+                if let Some(bytes) = linked_contract.get_deployed_bytecode_bytes() {
+                    known_contracts.insert(id.clone(), (abi.clone(), bytes.to_vec()));
+                }
+            }
+        }
+        if use_zk {
+            let zk_contracts =
+                zk_output.unwrap().with_stripped_file_prefixes(root).into_artifacts();
```

Comment:
Since we unwrap here, do we ever have `use_zk = true` and `zk_output = None` (or the opposite)? In that case we could just set `use_zk` to `zk_output.is_some()`, no?